vllm/core/scheduler.py

```diff
@@ -990,7 +1033,9 @@ def schedule(self) -> Tuple[List[SequenceGroupMetadata], SchedulerOutputs]:
             # It assumes the scheduled_seq_groups is ordered by
             # prefill < decoding.
             is_prompt = seq_group.is_prefill()
-            seq_group_metadata = SequenceGroupMetadata(
+
+            seq_group_metadata = seq_group_metadata_cache.get_object()
+            seq_group_metadata.__init__(
```

Comment:
It may be more straightforward to use a single API for this, such as

```suggestion
            seq_group_metadata = seq_group_metadata_cache.new_object(
```

Reply:
Refactored to a single API call in both cases